src/main/java/seedu/apollo/command/module/DeleteModuleCommand.java

```diff
@@ -76,6 +57,37 @@ public void execute(TaskList taskList, Ui ui, Storage storage, ModuleList module
         }
     }
 
+
+    /**
+     * Handles the deletion of a module from the module list.
+     *
+     * @param moduleList ModuleList to be deleted from.
+     * @param ui Ui to print messages.
+     * @throws ModuleNotFoundException If the module is not found.
+     */
+    private void handleSingleCommand(ModuleList moduleList, Ui ui) throws ModuleNotFoundException,
+            NumberFormatException {
+        String listParam = args[0];
```

Comment:
Perhaps you could further extract these into 2 different modules, deleteByIndex (for line 71 to 79) and deleteByModuleCode (line 80 to 88). Also more SLAP! and shorter methods